benchmarks/src/test/java/org/apache/druid/benchmark/query/SqlWindowFnBenchmark.java

```diff
@@ -0,0 +1,433 @@
+      walker.add(
+          descriptor,
+          FrameTestUtil.adapterToFrameSegment(
+              new QueryableIndexStorageAdapter(index),
+              FrameType.COLUMNAR,
+              descriptor.getId()
+          )
+      );
+    } else {
+      throw new IAE("Invalid storageType[%s]", storageType);
+    }
+  }
+
+  private static DruidOperatorTable createOperatorTable(final Injector injector)
+  {
+    try {
+      final Set<SqlOperatorConversion> extractionOperators = new HashSet<>();
+      extractionOperators.add(injector.getInstance(QueryLookupOperatorConversion.class));
+      final ApproxCountDistinctSqlAggregator countDistinctSqlAggregator =
+          new ApproxCountDistinctSqlAggregator(new HllSketchApproxCountDistinctSqlAggregator());
+      final Set<SqlAggregator> aggregators = new HashSet<>(
+          ImmutableList.of(
+              new DoublesSketchApproxQuantileSqlAggregator(),
+              new DoublesSketchObjectSqlAggregator(),
+              new HllSketchApproxCountDistinctSqlAggregator(),
+              new HllSketchApproxCountDistinctUtf8SqlAggregator(),
+              new ThetaSketchApproxCountDistinctSqlAggregator(),
+              new CountSqlAggregator(countDistinctSqlAggregator),
+              countDistinctSqlAggregator
+          )
+      );
+      return new DruidOperatorTable(aggregators, extractionOperators);
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @TearDown(Level.Trial)
+  public void tearDown() throws Exception
+  {
+    closer.close();
+  }
+
+  public void querySql(String sql, Blackhole blackhole)
+  {
+    final Map<String, Object> context = ImmutableMap.of(
+        QueryContexts.VECTORIZE_KEY, false,
+        QueryContexts.VECTORIZE_VIRTUAL_COLUMNS_KEY, false,
+        PlannerContext.CTX_ENABLE_WINDOW_FNS, true,
+        QueryContexts.MAX_SUBQUERY_BYTES_KEY, "auto"
+    );
+    try (final DruidPlanner planner = plannerFactory.createPlannerForTesting(engine, sql, context)) {
+      final PlannerResult plannerResult = planner.plan();
+      final Sequence<Object[]> resultSequence = plannerResult.run().getResults();
+      final Object[] lastRow = resultSequence.accumulate(null, (accumulated, in) -> in);
+      blackhole.consume(lastRow);
+    }
+  }
+
+  public void planSql(String sql, @Nullable Blackhole blackhole)
+  {
+    final Map<String, Object> context = ImmutableMap.of(
+        QueryContexts.VECTORIZE_KEY, false,
+        QueryContexts.VECTORIZE_VIRTUAL_COLUMNS_KEY, false,
+        PlannerContext.CTX_ENABLE_WINDOW_FNS, true,
+        QueryContexts.MAX_SUBQUERY_BYTES_KEY, "auto"
+    );
+    try (final DruidPlanner planner = plannerFactory.createPlannerForTesting(engine, sql, context)) {
+      final PlannerResult plannerResult = planner.plan();
+      if (blackhole != null) {
+        blackhole.consume(plannerResult);
+      }
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
```

Comment:
I wonder if these are the defaults; but anyway it could be set at class level as well

Reply:
moved to class level